src/main/scala/interfaces/Verification.scala

```diff
@@ -148,16 +149,55 @@ case class SiliconMappedCounterexample(
   val interpreter: ModelInterpreter[ExtractedModelEntry, Seq[ExtractedModelEntry]] = GenericDomainInterpreter(converter)
 
   override lazy val toString: String = {
-    val buf = converter.modelAtLabel
+    val extractedModel = converter.extractedModel
+    val buf_models = converter.modelAtLabel
       .map { case (label, model) => s"model at label $label:\n${interpret(model).toString}"}
       .mkString("\n")
-    s"$buf\non return: \n${interpret(converter.extractedModel).toString}"
+    val (buf_domains, buf_non_domain_functions) = functions_toString()
+    s"$buf_models\non return: \n${interpret(extractedModel).toString} $buf_domains $buf_non_domain_functions"
   }
   private def interpret(t: ExtractedModel): ExtractedModel =
     ExtractedModel(t.entries.map{ case (name, entry) => (name, interpret(entry)) })
   private def interpret(t: ExtractedModelEntry): ExtractedModelEntry = interpreter.interpret(t, Seq())
 
+  private def functions_toString() : (String, String) = {
+    //extractedModel.entries should be a bijection so we can invert the map
+    //and we only care about ref entries
+    val inv_entries = converter.extractedModel.entries.flatMap{ case (name, entry) => 
+      entry match {
+        case RefEntry(sym_name, _) => Some(sym_name -> name)
+        case NullRefEntry(sym_name)=> Some(sym_name -> name)
+        case _ => None
+      }}
+
+    val replace_domain_functions = converter.domains.map{ case DomainEntry(names, types, functions) => 
+      DomainEntry(names, types, replace_sym_names(functions, inv_entries))}
+    var buf_domains = replace_domain_functions.map(domain => s"${domain.toString()}").mkString("\n")
```

Comment:
`.map(domain => s"${domain.toString()}")` should just be `.map(domain => domain.toString())`? Same below.